src/mixins/object_geometry.mixin.js

```diff
@@ -88,10 +88,29 @@
      * The coords are returned in an array.
      * @return {Array} [tl, tr, br, bl] of points
      */
-    getCoords: function(absolute, calculate) {
+    getCoords: function (absolute, calculate) {
       return arrayFromCoords(this._getCoords(absolute, calculate));
     },
 
+    /**
+     * @returns {{ offset: fabric.Point, blur: fabric.Point }}
+     */
+    getShadowOffsets: function () {
+      var blur = this.shadow.blur;
+      var shadowOffset = new fabric.Point(this.shadow.offsetX, this.shadow.offsetY),
+          blurOffset = new fabric.Point(blur, blur);
+      if (!this.shadow.nonScaling) {
+        var scaling = this.getObjectScaling(),
```

Comment:
i think that on top of getObjectScaling you also need the zoom level.
Zoom will grow the shadow no matter what.

Reply:
I think you are right, pushed
[76a142d](https://github.com/fabricjs/fabric.js/pull/7698/commits/76a142d8874ab2c3920e34b23b292a99eda49594)